### PR TITLE
[Merged by Bors] - feat: the one-line-align linter

### PR DESCRIPTION
**Pull request by adomani**
This linter makes sure that `#align` statements span a single line.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Linter/Lint.lean [hunk 1]
```diff
@@ -108,4 +108,45 @@ def dupNamespace : Linter where run := withSetOptionIn fun stx => do
...
+spanning more than one line.
+-/
+register_option linter.oneLineAlign : Bool := {
+  defValue := true
+  descr := "enable the one-line align linter"
+}
+
+namespace OneLineAlignLinter
+
+open Lean Parser Elab Command Meta
```
Comment (grunweg) [inline Mathlib/Tactic/Linter/Lint.lean:131]: This suffices
```suggestion
open Lean

File: Mathlib/Tactic/Linter/Lint.lean [hunk 2]
```diff
@@ -108,4 +108,45 @@ def dupNamespace : Linter where run := withSetOptionIn fun stx => do
...
 initialize addLinter dupNamespace
 
-end Mathlib.Linter.DupNamespaceLinter
+end DupNamespaceLinter
+
+/-!
+#  `oneLineAlign` linter
+
+The `oneLineAlign` linter checks that `#align` statements span a single line,
+which is desirable as it allows them to be parsed by dumb parsers.
```
Comment (grunweg) [inline Mathlib/Tactic/Linter/Lint.lean:117]: What do you think?
```suggestion
which is desirable as it allows them to be parsed by very simple parsers.

**Review by grunweg — COMMENTED** [review]
The implementation looks good (two trivial comments); thanks for writing it.

I think this linter can be added as long as the aligns still exist (and be removed with them). I don't have a strong opinion on this point - let's see what the broader opinion is.
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by grunweg** [comment]
Can you merge master, please? I don't see why this should fail anything...

**Comment by adomani** [comment]
I think that it was conflicted with a section end in a test file.

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**